Documentation/mender-ci-cd.md

```diff
@@ -175,3 +174,90 @@ sudo mount.nfs4 $NFS_URI:/sstate-cache /mnt
+    MachineOptions = ["google-project=mender-gitlab-runners", "google-machine-type=n2-standard-16", "google-disk-size=100", "google-machine-image=https://www.googleapis.com/compute/v1/projects/mender-gitlab-runners/global/images/nested-virt-ubuntu-2204-jammy-v20221011", "google-tags=mender-qa-worker", "google-zone=northamerica-northeast1-b", "google-use-internal-ip=true", "google-scopes=https://www.googleapis.com/auth/devstorage.read_write,https://www.googleapis.com/auth/logging.write,https://www.googleapis.com/auth/monitoring.write,https://www.googleapis.com/auth/cloud-platform"]
+EOF
+```
+
+From the above, the only parameter that we need to manually modify per runner is
+`google-machine-type=n2-standard-16`. See more info below.
+
+Next, use `gitlab-runner register` to register runners. For each runner there are three key
+parameters to configure:
+* `--tag-list`: keywords to identify the kind of jobs that the runner will take. These map with the
+  CI jobs that we define in our pipelines and can be later edited (and moved around) using GitLab
+  UI.
+* `--name`: visible runner name as seen from GitLab UI. It cannot be modified after registration.
+  This is a name that represents the kind of machines that the runner will launch, not the kind of
+  jobs that the runner will accept.
+* `google-machine-type=` in `MachineOptions`: exact machine name to launch in Google Cloud Platform.
+
+
+For example, to register a runner with N2 16 cores machines to run client acceptance tests.
+
+* First execute `gitlab-runner` command:
+
+```
+sudo gitlab-runner register --non-interactive \
+    --template-config /etc/gitlab-runner/mender-runner-template-config.toml \
+    --registration-token <REDACTED> \
+    --access-level="not_protected" \
+    --locked="false" \
+    --run-untagged="false" \
+    --tag-list "mender-qa-worker-client-acceptance-tests" \
+    --name mender-runner-n2-standard-16
+```
+
+* Then manually edit `/etc/gitlab-runner/config.toml` replacing the value of `google-machine-type=`
+  for the desired one.
+
+
+#### GitLab runners in Mender CI infra
+
+This table summarizes the existing runners at the time of writing this lines.
+
+NOTE: Check GitLab UI and/or `/etc/gitlab-runner/config.toml` on the master's machine for the source
+of truth.
+
+| Runner's name                | GPC machine type | Pipeline tags |
+| ---------------------------- | ---------------- | ------------------------------------------ |
+| mender-runner-n2-standard-16 | n2-standard-16   | mender-qa-worker-integration-tests,mender-qa-worker-client-acceptance-tests |
+| mender-runner-n2-highcpu-16  | n2-highcpu-16    | mender-qa-worker-backend-integration-tests |
+| mender-runner-n2-standard-8  | n2-standard-8    |                                            |
+| mender-runner-n2d-standard-8 | n2d-standard-8   |                                            |
+| mender-runner-n2d-standard-4 | n2d-standard-4   | mender-qa-worker-generic                   |
```

Comment:
Maybe I'm gaping over too much in this PR, but should we consider a single CPU instance as well? I'm thinking of replacing public runners with our own runners, now that minutes are getting more expensive. Most of them don't need multiple cores.

Reply:
Sounds like a good idea to me also :smiley_cat:

Reply:
Sure, I will use one core and see.

The thing is that we can _move_ `mender-qa-worker-generic` tag to a lower end runner using GitLab UI without changing the pipeline definition. But I will anyway update this table.